src/Service/PluginPaymentMethodsService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2023 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Shopware\Service;
+
+use Adyen\AdyenException;
+use Adyen\Shopware\Handlers\CardsPaymentMethodHandler;
+use Adyen\Shopware\Provider\AdyenPluginProvider;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+
+class PluginPaymentMethodsService
+{
+    /** @var AdyenPluginProvider */
+    protected $adyenPluginProvider;
+
+    /** @var EntityRepositoryInterface */
+    protected $paymentMethodRepository;
+
+    public function __construct(
+        AdyenPluginProvider $adyenPluginProvider,
+        EntityRepositoryInterface $paymentMethodRepository
+    ) {
+        $this->adyenPluginProvider = $adyenPluginProvider;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    public function getPluginPaymentMethods(string $identifier = null): array
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(
+            new EqualsFilter('pluginId', $this->adyenPluginProvider->getAdyenPluginId())
+        );
+
+        if (isset($identifier)) {
+            $criteria->addFilter(
+                new EqualsFilter('pluginId', $this->adyenPluginProvider->getAdyenPluginId())
+            );
+        }
+
+        $paymentMethods = $this->paymentMethodRepository
+            ->search($criteria, Context::createDefaultContext())
+            ->getEntities();
+
+        return $paymentMethods->getElements();
+    }
+
+    /**
+     * @param string $paymentMethod
+     * @return string
+     * @throws AdyenException
+     */
+    public function getHandlerIdentifierFromTxVariant(string $paymentMethod): string
+    {
```

Comment:
this method will not catch edge case variants like `visa_amazonpay`, we might end up going down a rabbit hole trying to map all tx_variants in order to get identify adyen payments.
What do you think @candemiralp ?

Reply:
You are totally right, let's discuss it on a call.